logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -425,12 +473,13 @@ class DummyOutput2 < LogStash::Outputs::DroppingDummyOutput; end
       Thread.abort_on_exception = true
 
       @t = Thread.new do
```

Comment:
I am not sure how to properly refactor this, there is also the `@abort_on_exception = Thread.abort_on_exception` and this is all in a `before :each`.  ... 
I'd prefer to defer that into another issue.